phac-nml/irida-next
# Attachments concatenation UI modal

## What does this PR do and why?
This is a continuation of #294 and includes the following:
- The `Concatenate Files` button is disabled until 2 or more files are selected.
- The `Concatenate Files` modal includes a description.
- The `Concatenate Files` modal stays open during an error and displays that error in an alert within the modal.
- The selected files are listed within the new modal description.

## Screenshots or screen recordings
![Screenshot from 2023-11-29 12-58-55](https://github.com/phac-nml/irida-next/assets/325703/bfcf755f-b563-40d2-88ec-1336f350b290)
![Screenshot from 2023-11-29 13-00-34](https://github.com/phac-nml/irida-next/assets/325703/b98b64a4-96d0-477b-a33e-08135a5c892f)

## How to set up and validate locally
1. Navigate to sample.
2. Upload two or more files of the same type.
3. Check the files to be concatenated.
4. Click the `Concatenate Files` button.
5. Fill in a base name.
6. Click the `Concatenate` button.
7. Verify the original & concatenated file.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
